compiler/codegen/OMRCodeGenerator.cpp

```diff
@@ -391,6 +400,47 @@ void OMR::CodeGenerator::lowerTrees()
       TR_ASSERT(node->getVisitCount() != visitCount, "Code Gen: error in lowering trees");
       TR_ASSERT(node->getReferenceCount() == 0, "Code Gen: error in lowering trees");
 
+      if (node->getOpCodeValue() == TR::BBStart)
+         {
```

Comment:
Should these operations be guarded with
`if (comp()->getOption(TR_SplitWarmAndColdBlocks) && !comp()->compileRelocatableCode())` ?

Reply:
Put all the code in OMRCodeGenerator.cpp into a separate function and protected with getOption(TR_SplitWarmAndColdBlocks). The code actually supposed to work even with AOT (I think).

Reply:
I don't think it's the case in OMR, but in J9, the cold and warm sections are only split if `fej9()->needsContiguousCodeAndDataCacheAllocation()` is false; however, for relocatable compiles this returns true. I don't think J9's AOT supports discontiguous code sections.

Reply:
Will use `!comp()->compileRelocatableCode() `for this PR and we can discuss enabling with AOT in a separate issue.